### ISSUE TITLE
TelescopeMatching resets after TelescopeSelection

### ISSUE THREAD

**Issue by OkanEsen** [nvim-telescope/telescope.nvim#233]
The matching highlights reset after the selection is moved.

See: https://i.imgur.com/uYzUCmo.mp4

Shouldn't matter but these are the highlight groups:
``` vim
TelescopeMatching xxx gui=bold guifg=#83a598 guibg=none
TelescopeSelection xxx gui=bold guifg=#ebdbb2 guibg=#1d2021
```

**Comment by rockerBOO** [comment]
This also clears the devicon colors too.

**Comment by tjdevries** [comment]
Yeah, I'm not sure if I want to implement the fix here or if buffer highlights should be fixed in core to retain previous highlights when overridden. We will have to see.

**Comment by tjdevries** [comment]
@bfredl do is there anything I can do to solve this problem:

1. Namespace 1 applies highlights on line X
2. Namespace 2 applies highlights on lin X
3. Clear namespace 2.
4. Now there are no highlights.

**Comment by Conni2461** [comment]
Fixed in #355
